### PR TITLE
Code path normalize opencv 2

### PR DESCRIPTION
**Pull request by brunoais**
Remake of #119

### CODE REVIEW

**Comment by brunoais** [comment]
... #136 broken this one... Will fix later when I can.

**Comment by brunoais** [comment]
I'll test this when I can.
The previous commits were manually tested.... Oh well....

**Comment by YukihoAA** [comment]
I added some commits to fix some bugs and not unified names.

it works fine with linux and windows.

File: src/main.cpp [hunk 1]
```diff
@@ -208,83 +208,95 @@ bool validate_format_extension(std::wstring ext_w) {
...
-void debug_show_opencv_formats()
+void display_supported_formats()
 {
-	std::cout << "This is a list of supported formats, it depends on which formats opencv has been built with." << std::endl ;
-	for (auto const& x : opencv_formats)
+	std::cout
+	#ifdef HAVE_OPENCV
+			 << " [With OpenCV] This list depends on which formats opencv has been built with."
+	#else
+			 << " [Without OpenCV] (recompile to enable), Only the default formats can be used."
```
Comment (brunoais) [inline src/main.cpp]: > nable), Only the default
I recommend either replacing the comma (`,`) with a stop (`.`) or make the `O` in "Only" small.

File: src/main.cpp [hunk 2]
```diff
@@ -208,83 +208,95 @@ bool validate_format_extension(std::wstring ext_w) {
...
-void debug_show_opencv_formats()
+void display_supported_formats()
 {
-	std::cout << "This is a list of supported formats, it depends on which formats opencv has been built with." << std::endl ;
-	for (auto const& x : opencv_formats)
+	std::cout
+	#ifdef HAVE_OPENCV
+			 << " [With OpenCV] This list depends on which formats opencv has been built with."
+	#else
+			 << " [Without OpenCV] Only the default formats can be used (recompile to use OpenCV)."
```
Comment (brunoais) [inline src/main.cpp]: That works too.

**Pull request merged**